specs/rating/rating.md

```diff
@@ -0,0 +1,162 @@
+
+---
+
+### API
+
+Extends [RadioGroup](../radio-group/radio-group.spec.md).
+
+_Component name:_
+
+- `fast-rating`
+
+_Attributes:_
+
+- `disabled` - A boolean to remove interaction with the component.
+- `readonly` - A boolean to remove interaction with the component.
+- `value` - The current value of the item selected.
+- `mode` - enum
+  - multiple: This is the default mode. A mode where multiple items are highlighted for rating.
+  - single: A mode where the a user chooses only one item to rate.
+
+_Slots:_
+
+- `default` - Default slot is the slotted items.
+- `label` - Optional label. Some examples would be to indicate rating counts, or average ratings.
+
+### Anatomy and Appearance
+
+The Rating component template is extended from RadioGroup.
+
+_Template:_
+
+```html
+<template
+    role="radiogroup"
+    aria-disabled="${x => x.disabled}"
+    aria-readonly="${x => x.readOnly}"
+    @click="${(x, c) => x.clickHandler(c.event as MouseEvent)}"
+    @keydown="${(x, c) => x.keydownHandler(c.event as KeyboardEvent)}"
+    @focusout="${(x, c) => x.focusOutHandler(c.event as FocusEvent)}"
+    @mouseout="${(x, c) => x.onMouseout(c.event as MouseEvent)}"
+>
+    <div
+        class="positioning-region ${x =>
+            x.orientation === Orientation.horizontal ? "horizontal" : "vertical"}"
+        part="positioning-region"
+    >
+        <slot
+            ${slotted({
+                property: "slottedRadioButtons",
+                filter: elements("[role=radio]"),
+            })}
+        ></slot>
+    </div>
+    <slot name="label"></slot>
+</template>
+```
+
+## Implementation
+
+```html
+<fast-rating>
+    <fast-rating-item value="1"></fast-rating-item>
+    <fast-rating-item value="2"></fast-rating-item>
+    <fast-rating-item value="3"></fast-rating-item>
+    <fast-rating-item value="4"></fast-rating-item>
+    <fast-rating-item value="5"></fast-rating-item>
+</fast-rating>
+```
+
+### Accessibility
+
+Keyboard interaction
+
+- Tab and Shift + Tab will move focus in and out of the rating component. When focus moves onto the rating component, the first item is higlighted and selected.
+- Right and Down Arrow will move focus to the next rating item in the group. As focused is moved to the next item, both the focused item and the previous item are visually hilighted, but the the focused item is only selected.
+- Left and Up Arrow will move focus to the previous radio button in the group, removing the visual highlight from the previously focused item.
+
+
+Assistive Technology
+
+- When focused is on the component, the reader will announce the component, the current item and the the total amount of items. Example: "rating, 1 of 5 stars".
+As focused is moved to the next item, the reader will read out the current value and the amount of total items. Example: "2 of 5 stars, 3 of 5 stars, 4 of 5 stars, 5 of 5 stars".
+
+
+
+# Rating item
+
+## Overview
+The `fast-rating-item` is to used inside the `fast-rating` component.
```

Comment:
I don't love the `item` portion here. I'm curious - what makes up a "rating"? Is it the group together or the individual thing? If it's the group, does "item" make sense? It's better than something specific like "star" as that modifies the type of rating, but I'm curious if we can come up with a better semantic than this.

Reply:
Also, if this is just a radiogroup and radio - perhaps this only needs to be a `fast-components` implementation rather than something in foundation?

Reply:
My thought would be that the rating group component makes up the rating system. Since it will handle the items and the items could be any element, that has a role of radio. I went with the group and items system to follow the patterns of the other components that have a group and items, like menu and breadcrumb. But that does not mean Rating has to be the same. Just wanted to follow that pattern.
I did initially investigate just to have the Rating component and the template would use a repeat directive. I am happy to continue this approach if it makes more sense.  
I also had a similar thought that anyone can just use the RadioGroup and Radio and build their own rating system. But in this case, I guess one can say FAST Rating has already done the work for them. Maybe? Because it would have the functionalities to make it work as a rating system, accessibility, and other features already built.